web-common/src/features/models/workspace/inspector/References.svelte

```diff
@@ -84,8 +74,12 @@
                   : 'model'}/{reference?.resource?.meta?.name?.name}"
                 class="ui-copy-muted grid justify-between gap-x-2 {query_reference_trigger} pl-4 pr-4"
                 style:grid-template-columns="auto max-content"
-                on:focus={focus(reference.reference)}
-                on:mouseover={focus(reference.reference)}
+                on:focus={() => {
+                  eventBus.emit("highlightSelection", [reference.reference]);
+                }}
+                on:mouseover={() => {
+                  eventBus.emit("highlightSelection", [reference.reference]);
+                }}
```

Comment:
**Issue**: The new event handlers for focus and mouseover events emit a "highlightSelection" event without checking if the reference is valid. This could potentially lead to emitting events with undefined or null values, which might cause runtime errors or unexpected behavior in other parts of the application that listen to this event. <br> **Fix**: Add null checks before emitting the event to ensure that the reference is valid. This will prevent potential errors in other parts of the application. <br> **Code Suggestion**: 
 ```
 @@ -77,5 +77,5 @@
 -                on:focus={() => {
 -                  eventBus.emit("highlightSelection", [reference.reference]);
 -                }}
 -                on:mouseover={() => {
 -                  eventBus.emit("highlightSelection", [reference.reference]);
 +                on:focus={() => reference && eventBus.emit("highlightSelection", [reference.reference])}
 +                on:mouseover={() => reference && eventBus.emit("highlightSelection", [reference.reference])}
 ```

Reply:
The types don't suggest that reference can be undefined. I think this is being thrown off by the fact that the `?` operator is being used unnecessarily in other parts of this file.